박민영/MST (최소 스패닝 트리)/백준/1197.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * @author Minyeong Park
+ * @date 2023.03.27.
+ * 최소 스패닝 트리
+ * 최소 신장 트리 원리 참고 : https://sskl660.tistory.com/72
+ * 문제 링크 : https://www.acmicpc.net/problem/1197
+ */
+
+
+import java.io.*;
+import java.util.*;
+
+class Edge {
+    int start;
+    int end;
+    int weight;
+
+    public Edge(int start, int end, int weight) {
+        this.start = start;
+        this.end = end;
+        this.weight = weight;
+    }
+}
+
+public class Main {
+    static int v, e;
+    static ArrayList<Edge> list;
+    static int[] parent;
+    public static void main(String[] args) throws IOException {
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        StringTokenizer st = new StringTokenizer(br.readLine());
+        v = Integer.parseInt(st.nextToken());
+        e = Integer.parseInt(st.nextToken());
+
+        list = new ArrayList<>();
+        parent = new int[v + 1];
+        for (int i = 0; i < e; i++) {
+            st = new StringTokenizer(br.readLine());
+            int a = Integer.parseInt(st.nextToken());
+            int b = Integer.parseInt(st.nextToken());
+            int c = Integer.parseInt(st.nextToken());
+            list.add(new Edge(a, b, c));
+        }
+
+        // 그래프 데이터를 가중치 기준으로 정렬하기
+        Collections.sort(list, new Comparator<Edge>() {
+            @Override
+            public int compare(Edge o1, Edge o2) {
+                return o1.weight - o2.weight;
+            }
+        });
+
+        // 부모 배열 자기 자신 인덱스로 초기화
+        for (int i = 1; i <= v; i++) {
+            parent[i] = i;
+        }
+
+        int weight_sum = 0;
+        // 가중치가 낮은 엣지부터 연결 시도하기
+        for (int i = 0; i < e; i++) {
+            Edge now = list.get(i);
+            if (find(now.start) == find(now.end))
+                continue;
+
+            union(now.start, now.end);
+            weight_sum += now.weight;
+        }
+        System.out.println(weight_sum);
+    }
+
+    static void union(int a, int b) {
+        a = find(a);
+        b = find(b);
+        if (a > b) {
+            parent[a] = b;
+        } else {
+            parent[b] = a;
+        }
+    }
+
+    static int find(int a) {
+        if (a == parent[a])
+            return a;
+        else {
+            return find(parent[a]);
+        }
+    }
```

Comment:
```suggestion
    static int find(int a) {
        if (a == parent[a]) return a;
        return parent[a] = find(parent[a]);
    }
```
else문은 불필요해보입니다!
그리고 재귀의 결과를 parent[a]에 대입해줌으로써 경로 압축을 해줄 수 있다고 알고 있습니다!

Reply:
아 헉 그 부분을 놓치고 그냥 풀었네요... 놓친 부분 짚어주셔서 감사합니다!!